ai-service/src/ai_service/embedder.py

```
@@ -20,10 +22,10 @@ def embed_texts(texts: list[str]) -> np.ndarray:
         A NumPy array of embeddings.
 
     Raises:
-        ValueError: If texts is empty or contains only empty strings.
+        EmbeddingError: If texts is empty or contains only empty strings.
     """
     if not texts or all(not text.strip() for text in texts):
-        raise ValueError("Cannot embed empty or whitespace-only texts")
+        raise EmbeddingError("Cannot embed empty or whitespace-only texts")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)